src/glass/src/app/pages/announce-page/announce-page.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  FormBuilder,
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { marker as TEXT } from '@biesbjerg/ngx-translate-extract-marker';
+import * as _ from 'lodash';
+import { BlockUI, NgBlockUI } from 'ng-block-ui';
+import validator from 'validator';
+
+import { translate } from '~/app/i18n.helper';
+import { NodesService } from '~/app/shared/services/api/nodes.service';
+
+const TOKEN_REGEXP = /^[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}$/i;
+
+@Component({
+  selector: 'glass-announce-page',
+  templateUrl: './announce-page.component.html',
+  styleUrls: ['./announce-page.component.scss']
+})
+export class AnnouncePageComponent implements OnInit {
```

Comment:
I know it's bikeshedding but may I ask why you named the component **Announce**PageComponent? - Announce sounds a bit... inappropriate to me. Why not something like `JoinClusterPageComponent` or similar?

Reply:
It's named announce because this is what happens in that stage, leader and follower announce each other.

Reply:
Ah yes, I understand what you mean. I'm still not a big fan of `announce` but I don't have a better idea currently... maybe `introduce`/`introduction`?

Reply:
Renamed to `RegisterPageComponent` and the route `#/join/register`.